kernel/kernel-api/src/main/java/io/delta/kernel/internal/clustering/ClusteringUtils.java

```diff
@@ -37,13 +35,4 @@ public static DomainMetadata getClusteringDomainMetadata(List<Column> clustering
         ClusteringMetadataDomain.fromClusteringColumns(clusteringColumns);
     return clusteringMetadataDomain.toDomainMetadata();
   }
-
-  /**
-   * Extract ClusteringColumns from a given snapshot. Return None if the clustering domain metadata
-   * is missing.
-   */
-  public static Optional<List<Column>> getClusteringColumnsOptional(SnapshotImpl snapshot) {
```

Comment:
yup, love to see this deleted! This is weird C style struct code ... why would we invoke a method **on** a snapshot ... when we could just have all the snapshot APIs in one place and invoke a method **from** a snapshot